### PR TITLE
feat(app): Add GET /deck_configuration notification support

### PR DESCRIPTION
**Pull request by mjhuff**
Closes [EXEC-263](https://opentrons.atlassian.net/browse/EXEC-263)

# Overview

This is the FE equivalent of #14986. Refactors GET /deck_configuration to use notifications.

# Test Plan
- First, push #14986 to a robot. You may verify notifications for deck config are emitted by following the test plan in that PR.
- Open the network tab and go to some page/modal with deck config. Verify there is no polling. 
- Alter the deck config. Depending on which deck config you're using, the exact manner in which the data is sent to the server varies, but you should at some point see generally 1-2 network requests to `deck_configuration`. 

# Changelog
- Add desktop app & ODD MQTT support to GET /deck_configuration.

# Risk assessment
low - this is probably the most straightforward of all routes. 

[EXEC-263]: https://opentrons.atlassian.net/browse/EXEC-263?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by TamarZanzouri — APPROVED** [review]
LGTM!

File: app/src/organisms/ProtocolSetupDeckConfiguration/index.tsx [hunk 1]
```diff
@@ -56,7 +54,7 @@ export function ProtocolSetupDeckConfiguration({
   ] = React.useState<boolean>(false)
 
   const mostRecentAnalysis = useMostRecentCompletedAnalysis(runId)
-  const { data: deckConfig = [] } = useDeckConfigurationQuery()
+  const { data: deckConfig = [] } = useNotifyDeckConfigurationQuery()
```
Comment (koji) [inline app/src/organisms/ProtocolSetupDeckConfiguration/index.tsx]: nit 

const deckConfig = useNotifyDeckConfigurationQuery()?.data ?? []

File: app/src/pages/Protocols/hooks/index.ts [hunk 2]
```diff
@@ -83,7 +83,7 @@ export const useRequiredProtocolHardwareFromAnalysis = (
 
   const robotType = FLEX_ROBOT_TYPE
   const deckDef = getDeckDefFromRobotType(robotType)
-  const { data: deckConfig = [] } = useDeckConfigurationQuery({
+  const { data: deckConfig = [] } = useNotifyDeckConfigurationQuery({
     refetchInterval: DECK_CONFIG_REFETCH_INTERVAL,
   })
```
Comment (koji) [inline app/src/pages/Protocols/hooks/index.ts]: nit
same as the above

**Review by koji — APPROVED** [review]
tested with a Flex and this branch worked as expected.
the changes look good to me.

I just left 2 comments, but they are just coding style.

**Pull request merged**